src/drivers/kinetis/i2c.c

```diff
@@ -0,0 +1,310 @@
+#include <MK64F12.h>
+#include <drivers/i2c.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/* SCL clock divider table for baudrate calculation */
+static const uint16_t SCL_LUT[] = {
+    20,  22,  24,  26,   28,   30,   34,   40,   28,   32,   36,   40,   44,   48,   56,   68,
+    48,  56,  64,  72,   80,   88,   104,  128,  80,   96,   112,  128,  144,  160,  192,  240,
+    160, 192, 224, 256,  288,  320,  384,  480,  320,  384,  448,  512,  576,  640,  768,  960,
+    640, 768, 896, 1024, 1152, 1280, 1536, 1920, 1280, 1536, 1792, 2048, 2304, 2560, 3072, 3840};
+
+/* global variable to store the channel parameters */
+volatile I2C_Channel i2c_chs[I2C_NUM_DEVICES];
+
+/* global variable to store the base pointers */
+static I2C_Type *i2c_bases[] = I2C_BASE_PTRS;
+
+void init_m_i2c(i2c_config_t config, uint32_t clk_f_hz) {
+    /* get the base pointer */
+    I2C_Type *base = i2c_bases[config.i2c_num];
+
+    /* get the irq pointers */
+    uint32_t i2c_irqs[] = I2C_IRQS;
+
+    /* route clocks to I2C module */
+    switch(config.i2c_num) {
+        case 0:
+            SIM->SCGC4 |= SIM_SCGC4_I2C0_MASK;
+            break;
+        case 1:
+            SIM->SCGC4 |= SIM_SCGC4_I2C1_MASK;
+            break;
+        case 2:
+            SIM->SCGC1 |= SIM_SCGC1_I2C2_MASK;
+            break;
+    }
```

Comment:
Just a note that we'll need to remove this later when we have overlays/device tree (if we get that far).  No change needed.